Jupyter_Notebooks/PreProcess.py

```diff
@@ -0,0 +1,279 @@
+import ipywidgets as widgets
+import numpy as np
+import sys
+import scipy.io
+sys.path.insert(1, 'C:/Users/joelb/pyCUFSM')
+import pycufsm.plotters as crossect
+def prevals():
+    springs = []
+    constraints = []
+    flag = np.array([1, 0, 0, 0, 1, 0, 0, 0, 0, 0])
+    props = np.array([[0, 29500, 29500, 0.3, 0.3, 29500 / (2 * (1 + 0.3))]])
+    nodes =np.array([[0, 2.5, 0.773, 1, 1, 1, 1, 1], [1, 2.5, 0, 1, 1, 1, 1, 1],
+                [2, 1.25, 0, 1, 1, 1, 1, 1], [3, 0, 0, 1, 1, 1, 1, 1],
+                [4, 0, 3, 1, 1, 1, 1, 1], [5, 0, 6, 1, 1, 1, 1, 1],
+                [6, 0, 9, 1, 1, 1, 1, 1], [7, 1.25, 9, 1, 1, 1, 1, 1],
+                [8, 2.5, 9, 1, 1, 1, 1, 1], [9, 2.5, 8.227, 1, 1, 1, 1, 1]])
+    elements = np.array([[0, 0, 1, 0.059, 0], [1, 1, 2, 0.059, 0], [2, 2, 3, 0.059, 0],
+                    [3, 3, 4, 0.059, 0], [4, 4, 5, 0.059, 0], [5, 5, 6, 0.059, 0],
+                    [6, 6, 7, 0.059, 0], [7, 7, 8, 0.059, 0], [8, 8, 9, 0.059, 0]])
+    return props, nodes, elements, springs, constraints, flag
+# def input(page, length_BC):
+#     props, nodes, elements, springs, constraints, flag = prevals()
+#     page, props, nodes, elements = PreProcess(m,n,e, props, nodes, elements, page, springs, constraints, flag)
+#     children = [page, page]
+#     tab = widgets.Tab()
+#     tab.children = children
+#     tab.set_title(0, 'Main Preprocessor')
+#     tab.set_title(1, 'Lengths and Boundary Conditions')
+#     return tab
+
+class Preprocess:
+
+    def wprops(self, props, m):
```

Comment:
Can you add comments at least at the beginning of each function saying what the function does? It's very difficult to follow the code in here.

Reply:
Yes this script isn't gonna be in the final version..didn't mean to add this to the main repo